### ISSUE TITLE
Invalid suggestion from needless_bool when condition spans multiple lines

### ISSUE THREAD

**Issue by dtolnay** [rust-lang/rust-clippy#3481]
```rust
fn main() {
    let b = false;
    let used_addr = 0u64;
    let used_ring_size = 0;

    let x = if b {
        unimplemented!()
    } else if used_addr
        .checked_add(used_ring_size)
        .map_or(true, |n| n % 2 == 0)
    {
        false
    } else {
        true
    };

    println!("{}", x);
}
```

```console
warning: this if-then-else expression returns a bool literal
  --> src/main.rs:8:12
   |
8  |       } else if used_addr
   |  ____________^
9  | |         .checked_add(used_ring_size)
10 | |         .map_or(true, |n| n % 2 == 0)
11 | |     {
...  |
14 | |         true
15 | |     };
   | |_____^
   |
   = note: #[warn(clippy::needless_bool)] on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_bool
help: you can reduce it to
   |
8  |     } else !used_addr
9  |         .checked_add(used_ring_size)
10 |         .map_or(true, |n| n % 2 == 0);
   |
```

Clippy's suggestion is not legal syntax:

```rust
} else !used_addr
    .checked_add(used_ring_size)
    .map_or(true, |n| n % 2 == 0);
```

---

*`clippy 0.0.212 (b2601be 2018-11-27)`*

**Comment by fkohlgrueber** [comment]
What should the suggestion be here? Like this?

```
fn main() {
    let b = false;
    let used_addr = 0u64;
    let used_ring_size = 0;

    let x = if b {
        unimplemented!()
    } else {
        !used_addr
            .checked_add(used_ring_size)
            .map_or(true, |n| n % 2 == 0)
    };
    println!("{}", x);
}
```

**Comment by flip1995** [comment]
Yes it seems the problem here is, that it makes the right suggestion, but it has to be put in a block (only in the case of an `else if`?):
```rust
} else bool_expr;
```
vs
```rust
} else {
    bool_expr
};
```

**Comment by g-bartoszek** [comment]
@dtolnay  Here is my attempt to fix it: #3680
